### PR TITLE
Add anchors to some guides and all module categories

### LINKED ISSUE #36240: References from module docs to docsite guide
##### ISSUE TYPE
 - Documentation Report

##### COMPONENT NAME
docs

##### ANSIBLE VERSION
v2.6

##### SUMMARY
I am looking for a simple way to reference the ACI guide from each individual ACI module (inside e.g. a note).l

I also would like to reference the ACI module index from the ACI guide, but there's currently no anchor added for internal references for each module-section.

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
This is required if we want to use *absolute* :ref: references instead of *relative* :doc: references.

This PR includes:
- Anchor for `aci_guide` and `network_guide`
- Anchors for `playbooks_vault` and `playbooks_environment`
- Anchors for module/plugin categories (e.g. `network_modules`)
- Anchors for module/plugin section (e.g. `aci_network_modules`)

This fixes #36240

This should be backported.

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
Various docs

##### ANSIBLE VERSION
v2.5

### CODE REVIEW

**Comment by gundalow** [comment]
I like having the ability to link to module_categories
@acozine This is something we could use to link to (say) the eos modules from the eos platform specific page

**Comment by dagwieers** [comment]
@gundalow I think we should move most if not all references as :ref: references (absolute) instead of :doc: references (relative). At least reorganizing documents will not lead to links turning up as just text.

We also need to rewrite a lot of dangling :doc: references, so now is the time to fix this.

File: lib/ansible/utils/module_docs_fragments/aci.py [hunk 1]
```diff
@@ -83,6 +71,5 @@ class ModuleDocFragment(object):
     type: bool
     default: 'yes'
 notes:
-- Please read :ref:`the ACI guide <aci_guide>` for more detailed information
-  on how to manage your ACI infrastructure using Ansible.
+- Please read :ref:`aci_guide` for more detailed information on how to manage your ACI infrastructure using Ansible.
```
Comment (gundalow) [inline lib/ansible/utils/module_docs_fragments/aci.py]: Does that work correctly?
Just checking this doesn't get escaped and ends up not being valid RST

**Comment by gundalow** [comment]
Assigning to Scott as it's his decision

**Review by dharmabumstead — APPROVED** [review]
Looks good to me.

**Comment by gundalow** [comment]
@dagwieers regarding
> @gundalow I think we should move most if not all references as `:ref:` references (absolute) instead of `:doc:` references (relative). At least reorganizing documents will not lead to links turning up as just text.
> We also need to rewrite a lot of dangling `:doc:` references, so now is the time to fix this

I believe @dharmabumstead may be working on similar at the moment, so just want to avoid duplicated work

File: lib/ansible/utils/module_docs_fragments/aci.py (hunk 1 above)
Comment (dagwieers) [reply lib/ansible/utils/module_docs_fragments/aci.py]: This works and the link gets replaced with the title.

**Comment by acozine** [comment]
+1 to using absolute references instead of `:doc:` references - either one requires maintenance, but the latter lets us move pages around in the filetree with less disruption. We'll need to weed out duplication of titles as/when we switch.

**Comment by acozine** [comment]
And a huge +1 (+10?) for creating/linking to `platform-name_network_modules` - this will be incredibly useful.

**Comment by gundalow** [comment]
@acozine is testing locally and will then merge

**Comment by acozine** [comment]
Ah, I see - the `platform-name_network_modules` links take you to the subheading for that platform on the `list_of_network_modules` page. A good step forward, thanks!

**Review by acozine — APPROVED** [review]

**Comment by nitzmahone** [comment]
Does not cherry-pick cleanly to stable-2.5

**Pull request merged**